### PR TITLE
Fix for VelocityTracker deprecation

### PR DESCRIPTION
**Pull request by Piinks**
Part of #74908

Adds dart fix for deprecated VelocityTracker constructor 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/fix_data.yaml [hunk 1]
```diff
@@ -11,6 +11,28 @@
...
+        index: 0
+        name: 'kind'
+        style: required_positional
+        argumentValue:
+          expression: 'PointerDeviceKind.touch'
+          requiredIf: "pointerDeviceKind == ''"
+          variables:
+            pointerDeviceKind:
+              kind: 'fragment'
+              value: 'arguments[0]'
```
Comment (Piinks) [inline packages/flutter/lib/fix_data.yaml:34]: I've tried this a couple different ways now. The rename works fine, but adding the parameter does not work. I can see there is an error in the file (fix_data.yaml is red-underlined in the project directory), but I don't know where the error is (https://github.com/dart-lang/sdk/issues/44890).

Comment (Piinks) [reply packages/flutter/lib/fix_data.yaml:34]: Error does show up in analysis panel. So that is better, but I still can't seem to figure this fix out.

**Comment by Piinks** [comment]
I still can't quite figure this fix. @goderbauer @bwilkerson WDYT? I've tried a few different approaches now, the `rename` works OK, it's the `addParameter` where the old constructor had a default value that I can't get right. 

I am wondering if the `addParameter` is having issue with the fact that I am adding a parameter that already exists (`kind`), I just need to sub in `PointerDeviceKind.touch` when the default value is being used from the old empty constructor. 🤔

**Comment by bwilkerson** [comment]
Sorry it took me so long to respond. I believe that this is not working because of a bug in the data-driven fixes support. It isn't recognizing the constructor invocation as a valid place to apply the `addParameter` change. I'm working on a fix now and will update here when I know more.

**Comment by Piinks** [comment]
Thanks so very much @bwilkerson! 😄

**Comment by bwilkerson** [comment]
Ok, the fix has landed in the sdk repo as https://dart-review.googlesource.com/c/sdk/+/184205.

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**